spring-kafka/src/test/java/org/springframework/kafka/support/serializer/StringOrBytesSerializerTests.java

```diff
@@ -51,6 +53,9 @@ void test() {
 		Map<String, Object> configs = Collections.singletonMap("serializer.encoding", "UTF-16");
 		serializer.configure(configs, false);
 		assertThat(KafkaTestUtils.getPropertyValue(serializer, "stringSerializer.encoding")).isEqualTo(StandardCharsets.UTF_16);
+		assertThat(serializer.serialize("null", null)).isNull();
+		assertThatExceptionOfType(IllegalStateException.class).isThrownBy(() -> serializer.serialize("ex", 0))
```

Comment:
There is a dedicated `assertThatIllegalStateException()` factory method:
```suggestion
		assertThatIllegalStateException().isThrownBy(() -> serializer.serialize("ex", 0))
```